core/controllers/collection_editor.py

```diff
@@ -0,0 +1,206 @@
+
+    def _require_valid_version(self, version_from_payload, collection_version):
+        """Check that the payload version matches the given collection version.
+        """
+        if version_from_payload is None:
+            raise base.BaseHandler.InvalidInputException(
+                'Invalid POST request: a version must be specified.')
+
+        if version_from_payload != collection_version:
+            raise base.BaseHandler.InvalidInputException(
+                'Trying to update version %s of collection from version %s, '
+                'which is too old. Please reload the page and try again.'
+                % (collection_version, version_from_payload))
+
+    @require_editor
+    def put(self, collection_id):
+        """Updates properties of the given collection."""
+
+        collection = collection_services.get_collection_by_id(collection_id)
+        version = self.payload.get('version')
+        self._require_valid_version(version, collection.version)
+
+        commit_message = self.payload.get('commit_message')
+        change_list = self.payload.get('change_list')
+
+        try:
+            collection_services.update_collection(
+                self.user_id, collection_id, change_list, commit_message)
+        except utils.ValidationError as e:
+            raise self.InvalidInputException(e)
+
+        # Retrieve the updated collection.
+        collection_dict = (
+            collection_services.get_learner_collection_dict_by_id(
+                collection_id, self.user_id, allow_invalid_explorations=True))
+
+        # Send the updated collection back to the frontend.
+        self.values.update(collection_dict)
+        self.render_json(self.values)
+
+
+class CollectionRightsHandler(CollectionEditorHandler):
+    """Handles management of collection editing rights."""
+
+    @require_editor
+    def put(self, collection_id):
+        """Updates the editing rights for the given collection."""
+        collection = collection_services.get_collection_by_id(collection_id)
+        version = self.payload.get('version')
+        _require_valid_version(version, collection.version)
+
+        # TODO(bhenning): Implement other rights changes here.
+        is_public = self.payload.get('is_public')
+
+        if is_public is not None:
+            if is_public:
+                try:
+                    collection.validate(strict=True)
+                    collection_services.validate_collection_public_explorations(
+                        collection)
+                except utils.ValidationError as e:
+                    raise self.InvalidInputException(e)
+
+                collection_services.publish_collection_and_update_user_profiles(
+                    self.user_id, collection_id)
+                collection_services.index_collections_given_ids([
+                    collection_id])
+            else:
+                raise self.InvalidInputException(
```

Comment:
Hmm, good point. Updated to allow admins to unpublish (though it's not clear how they would even do that).